torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -531,45 +528,25 @@ class CppMicroGemmAMX(CppMicroGemm):
     alignas(4096) {{input_t}} dequantized_B_buf[buf_size];
     {%- endif %}
 
-    const auto b_tile_ptr_stride = ldb * {{vnni_size}};
-
-    auto load_B_row = [&]({{input2_t}}* {{restrict_keyword}} src, {{input_t}}* {{restrict_keyword}} dst) {
-        auto b_int8 = at::vec::Vectorized<int8_t>::loadu(src, static_cast<int64_t>(32));
-        auto b_bf16 = at::vec::convert<{{input_t}}>(b_int8);
-        b_bf16.store(dst);
-    };
-
-    auto load_B_tile = [&]({{input2_t}}* B_ptr, int idx, int num_b_rows) {
-        {{input_t}}* base_addr = dequantized_B_buf + idx;
-        {{kernel.unroll_pragma(8)}}
-        for (int i = 0; i < num_b_rows; i++) {
-            load_B_row(
-                B_ptr + i * b_tile_ptr_stride,
-                base_addr + i * 32
-            );
-        }
-    };
     auto load_dequantized_B = [&](int n) {
         // Load a tile of B & cache it in L1D.
-        {{kernel.unroll_pragma(4)}}
-        for (int k = 0; k < K; k += {{block_k}}) {
-            int num_b_rows = (k < last_k_offset) ? 16 : tail_k_size;
-            {{kernel.unroll_pragma(2)}}
-            for (int tile_col = 0; tile_col <= 1; tile_col++) {
-                load_B_tile(
-                    const_cast<{{input2_t}}*>(B) + n + k * ldb + tile_col * {{16 * vnni_size}},
-                    (k / {{block_k // 2}} + tile_col) * num_elements_per_b_tile,
-                    num_b_rows
-                );
-            }
+        // The assumption here is that if N would be a multiple of block_n,
+        // then B would be some consecutive tiles sized [k, block_n] each, so the
+        // first element of a subsequent [K, block_n] tile would be right after the last
+        // element of the previous tile sized [K x block_n]
+        auto base_idx = n * K * {{block_n}};
+        for (int idx = 0; idx < buf_size; idx += {{block_k}}) {
```

Comment:
@leslie-fang-intel, since `block_k` is always 32 for int8 case, I removed the `if-else` conditional in the patch.

Reply:
- I still feels better to use the 32 here which seems more readable.
- What's the case when `buf_size (K * {{block_n}})` is not divisible by `block_k(32)`? Are the load and store with
memory access out of bounds implicitly?

Reply:
`K` is always a multiple of 2, and `Nr` is either 16, 32 or 48.
So, `K * Nr` is always a multiple of 32.